src/main/java/com/bootcamp/demo/data_access/AccountRepository.java

```diff
@@ -29,14 +30,49 @@ public String add(Account account) throws ExecutionException, InterruptedExcepti
         return futureTransaction.get();
     }
 
+    @Override
+    public Account findOne(String username) throws ExecutionException, InterruptedException {
+        CollectionReference collectionReference = this.getAccountsCollection();
+        DocumentReference documentReference = collectionReference.document(username);
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot documentSnapshot = future.get();
+
+        if (documentSnapshot.exists()) {
+            Map<String, Object> data = documentSnapshot.getData();
+
+            String name = data.get("name").toString();
+            String email =  data.get("email").toString();
+            String phoneNumber =  data.get("phoneNumber").toString();
+            String address =  data.get("address").toString();
+            String dateOfBirth =  data.get("dateOfBirth").toString();
+
+            return new Account(name, email, username, null, phoneNumber, address, dateOfBirth);
```

Comment:
Instead of having a constructor with a lot of parameters we can use builder pattern. This way we will also avoid setting unnecessary fields